phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4977,25 +4994,27 @@ public void removeConnection(PhoenixConnection connection) throws SQLException {
         if (returnSequenceValues) {
             ConcurrentMap<SequenceKey,Sequence> formerSequenceMap = null;
             synchronized (connectionCountLock) {
-                if (--connectionCount <= 0) {
-                    if (!this.sequenceMap.isEmpty()) {
-                        formerSequenceMap = this.sequenceMap;
-                        this.sequenceMap = Maps.newConcurrentMap();
+                if(!connection.isInternalConnection()) {
+                    if (connectionCount + internalConnectionCount - 1 <= 0) {
```

Comment:
hmm, can you explain this condition? Why does internalConnectionCount come into the picture here?

Reply:
I was trying to leave this condition the same as before thus combining both connection counts.  This appears to me to be sequence handling for no connection open.